libs/designsystem/src/lib/components/checkbox/checkbox.component.scss

```
@@ -1,52 +1,102 @@
 @import '../../scss/utils';
 
+$border-radius: 6px;
+$checkbox-icon-size: size('m');
+
 :host {
-  display: inline-block;
-  vertical-align: middle;
-}
+  display: block;
 
-.checkbox {
-  --border-color: #{get-color('primary-contrast')};
-}
+  &.attention-level1 ion-checkbox {
+    --checkmark-color: #{get-color('black')};
+    --background-checked: #{get-color('success')};
+    --border-color-checked: #{get-color('success')};
 
-.primary {
-  --checkmark-color: #{get-color('primary-contrast')};
-  --background-checked: #{get-color('primary')};
-  --border-color-checked: #{get-color('primary')};
-}
+    &:active {
+      --background-checked: #{get-color('success-shade')};
+      --border-color-checked: #{get-color('success-shade')};
+    }
+  }
 
-.secondary {
-  --checkmark-color: #{get-color('secondary-contrast')};
-  --background-checked: #{get-color('secondary')};
-  --border-color-checked: #{get-color('secondary')};
-}
+  &.attention-level2 ion-checkbox {
+    --checkmark-color: #{get-color('white')};
+    --background-checked: #{get-color('black')};
+    --border-color-checked: #{get-color('black')};
 
-.tertiary {
-  --checkmark-color: #{get-color('tertiary-contrast')};
-  --background-checked: #{get-color('tertiary')};
-  --border-color-checked: #{get-color('tertiary')};
-}
+    &:active {
+      --checkmark-color: #{get-color('white-shade')};
+      --background-checked: #{get-color('black-tint')};
+      --border-color-checked: #{get-color('black-tint')};
+    }
+  }
 
-.warning {
-  --checkmark-color: #{get-color('warning-contrast')};
-  --background-checked: #{get-color('warning')};
-  --border-color-checked: #{get-color('warning')};
-}
+  &.error ion-checkbox {
+    --border-color: #{get-color('danger')};
+  }
 
-.success {
-  --checkmark-color: #{get-color('success-contrast')};
-  --background-checked: #{get-color('success')};
-  --border-color-checked: #{get-color('success')};
-}
+  &.disabled {
```

Consider disabling through a `disabled` attribute as this would be semantically better
```suggestion
  &:disabled {
```

---

I can't get :disabled to work

---

Using [disabled] instead